integrations/kafka_connector/src/main/java/software/amazon/timestream/utility/DLQReporter.java

```diff
@@ -0,0 +1,117 @@
+     */
+    private final String dlqTopicName;
+
+    /**
+     * Error reporter
+     */
+    private final ErrantRecordReporter dlqReporter;
+
+    /**
+     * Kafka client: DLQ Publisher
+     */
+    private KafkaProducer<String, String> dlqPublisher;
+
+    /**
+     *
+     * @param config Timestream Sink connector Configuration
+     */
+    public DLQReporter(final TimestreamSinkConnectorConfig config, final ErrantRecordReporter reporter) {
+        this.dlqTopicName = config.getString(TimestreamSinkConstants.DLQ_TOPIC_NAME);
+        this.dlqReporter = reporter;
+        try {
+            this.dlqPublisher = new KafkaProducer<>(initiateDLQConfiguration(config));
+            LOGGER.info("DLQReporter: bootstrapServer: [{}], dlqTopicName: [{}]", config.getString(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG), dlqTopicName);
+        } catch (KafkaException ke) {
+            LOGGER.error("ERROR: while configuring DLQReporter kafka producer client", ke);
+        }
+    }
+
+    /**
+     * Method to initiate DLQ Publisher configuration
+     * @param config TimestreamSinkConnectorConfig
+     * @return DLQ Publisher configuration
+     */
+    private Properties initiateDLQConfiguration(final TimestreamSinkConnectorConfig config) {
+        final Properties dlqPubConfig = new Properties();
+        dlqPubConfig.put(CommonClientConfigs.CLIENT_ID_CONFIG, TimestreamSinkConstants.CONST_DLQ_CLIENT_ID + "-" + Math.random());
+        dlqPubConfig.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, config.getString(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG));
+        dlqPubConfig.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, config.getString(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG));
+        dlqPubConfig.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, config.getString(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG));
+        //Security: IAM Authentication based
+        dlqPubConfig.setProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_SSL.name);
+        dlqPubConfig.put(SaslConfigs.SASL_MECHANISM, TimestreamSinkConstants.DEFAULT_SASL_MECHANISM );
+        dlqPubConfig.put(SaslConfigs.SASL_JAAS_CONFIG, TimestreamSinkConstants.DEFAULT_SASL_JAAS_CONFIG);
+        dlqPubConfig.put(SaslConfigs.SASL_CLIENT_CALLBACK_HANDLER_CLASS, TimestreamSinkConstants.DEFAULT_SASL_CLIENT_CALLBACK_HANDLER);
+        LOGGER.info("DLQPublisher: initiateDLQPublisherProperties: [{}]", dlqPubConfig);
+        return dlqPubConfig;
+    }
+    /**
+     * Method to publish the rejected Timestream record to DLQ for further processing
+     *
+     * @param rejectedRecords: Rejected Timestream records
+     */
+    public void reportRejectedRecords(final List<RejectedRecord> rejectedRecords) {
+        LOGGER.trace("Begin::DLQPublisher::reportRejectedRecords");
+        for (final RejectedRecord record : rejectedRecords) {
+            final String rejRecordTxt = GSON.toJson(record.toString());
```

Comment:
for some reason if there was an exception processing the rejected records,  this could fail publishing the remaining rejected records in the DLQ.  Should there be an exception handling added here ?

Reply:
Thanks, added exception handling with in the loop so the rest of the records are not missed to be published to DLQ in case of of an issue with a single rejected record